### PR TITLE
Benches refactor

### PR DESCRIPTION
**Pull request by mamcx**
# Description of Changes

Moving out from the bench timings the creation of dbs because is expensive -in special for spacetime that creates folders vs sqlite just a file- and generates noise.

# API

 - [ ] This is a breaking change to the module API
 - [ ] This is a breaking change to the ClientAPI
 - [ ] This is a breaking change to the SDK API

*If the API is breaking, please state below what will break*

### CODE REVIEW

**Review by cloutiertyler — CHANGES_REQUESTED** [review]
Requesting changes so this doesn't actually merge. Need to look at the batch api.

**Comment by kulakowski** [comment]
> Requesting changes so this doesn't actually merge. Need to look at the batch api.

Do you want me to take a look at this?

**Comment by cloutiertyler** [comment]
> > Requesting changes so this doesn't actually merge. Need to look at the batch api.
> 
> Do you want me to take a look at this?

Yes please @kulakowski

File: crates/core/src/db/datastore/traits.rs [hunk 1]
```diff
@@ -280,7 +280,7 @@ pub struct TableDef {
 impl From<ProductType> for TableDef {
     fn from(value: ProductType) -> Self {
         Self {
-            table_name: "".to_string(),
+            table_name: format!("t_{:x}", calculate_hash(&value)),
```
[the remark this answers is not in the document]
Comment (mamcx) [reply crates/core/src/db/datastore/traits.rs:283]: This line will be unnecessary in the [schema rework PR](https://github.com/clockworklabs/SpacetimeDB/pull/243).

[the remark this answers is not in the document]
Comment (mamcx) [reply crates/core/src/db/datastore/traits.rs:283]: A quick check I think it could be empty, but this is only called from test or benches.

**Comment by kazimuth** [comment]
I've integrated many of the changes here into #289. In particular, I tried iter_batched, then switched to a custom criterion timing loop so that we don't have to recreate the database every iteration. I don't have your `insert_batched` internal datastore changes though.

**Comment by mamcx** [comment]
This PR have been made obsolete by https://github.com/clockworklabs/SpacetimeDB/pull/289

**Comment by mamcx** [comment]
This is PR is obsoleted by https://github.com/clockworklabs/SpacetimeDB/pull/289

**Pull request closed without merging**